### ISSUE TITLE
Crashes upon startup: fs.js:1172

### ISSUE THREAD

**Issue by OnkelTem** [kosmtik/kosmtik#102]
This is how I start it:

```
node index.js serve ../styles/openstreetmap-carto/project.yaml 
```

And the log:
```
[Core] No usable config file found in /root/.config/kosmtik.yml
[Core] Loading plugin from ../plugins/base-exporters/index.js
[Core] Loading plugin from ../plugins/hash/index.js
[Core] Loading plugin from ../plugins/local-config/index.js
[Core] Loading plugin from ../plugins/datasource-loader/index.js
[Core] Loading project from ../styles/openstreetmap-carto/project.yaml
[Core] PreviewServer started, you can browse http://127.0.0.1:6789
[Local Config] Unable to load local config from /home/osm/styles/openstreetmap-carto/localconfig.js
[Error: File not found: /home/osm/styles/openstreetmap-carto/localconfig.js]
[Core] Generating Mapnik XML…
[Core] Using mapnik version 3.0.0
[Core] Loading map…
[Core] Map ready
[Core] Loading map…
[Core] Map ready
fs.js:1172
    throw errnoException(err, 'watch');
          ^
Error: watch ENOSPC
    at exports._errnoException (util.js:746:11)
    at FSWatcher.start (fs.js:1172:11)
    at Object.fs.watch (fs.js:1198:11)
    at Array.<anonymous> (/home/osm/kosmtik/src/back/ProjectServer.js:22:12)
    at null.<anonymous> (/home/osm/kosmtik/src/back/StateBase.js:27:43)
    at null.<anonymous> (/home/osm/kosmtik/src/back/StateBase.js:37:13)
    at DataSourceLoader.patchMML (/home/osm/kosmtik/src/plugins/datasource-loader/index.js:36:15)
    at null.<anonymous> (/home/osm/kosmtik/src/back/StateBase.js:35:74)
    at done (/home/osm/kosmtik/src/plugins/local-config/index.js:14:23)
    at LocalConfig.patchMML (/home/osm/kosmtik/src/plugins/local-config/index.js:31:16)
```

versions:
```
root@5c9f30d1e05a:/home/osm/kosmtik# node -v
v0.12.9
root@5c9f30d1e05a:/home/osm/kosmtik# npm -v
3.5.2
root@5c9f30d1e05a:/home/osm/kosmtik# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.1 LTS
Release:        14.04
Codename:       trusty
```

**Comment by pnorman** [comment]
Are any of your drives full or near full?

**Comment by OnkelTem** [comment]
No, at least from what I see:

```
rootfs         497793216 413613956  58869740  88% /
none           497793216 413613956  58869740  88% /
tmpfs            8203228         0   8203228   0% /dev
tmpfs            8203228         0   8203228   0% /sys/fs/cgroup
/dev/sdb3      497793216 413613956  58869740  88% /etc/hosts
shm                65536         0     65536   0% /dev/shm
tmpfs            8203228         0   8203228   0% /proc/kcore
tmpfs            8203228         0   8203228   0% /proc/latency_stats
tmpfs            8203228         0   8203228   0% /proc/timer_stats
```

I should say that that I'm in a docker container. Dunno if that matters.

**Comment by OnkelTem** [comment]
I've figured out the reason. The issue triggers in `ProjectServer.js` at `fs.watch(...) { ... }` and commenting that out was fixing the problem. After some googling I found nice article https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers 
Tl;dr, you should increase fs.inotify.max_user_watches value (it was 65535 on my machine).

To set a new limit temporary:
```
$ sudo sysctl fs.inotify.max_user_watches=524288
$ sudo sysctl -p
```
To make a new limit permanent:
```
$ echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf
$ sudo sysctl -p
```

> You may also need to pay attention to the values of max_queued_events and max_user_instances if Listen keeps on complaining.

**Comment by pnorman** [comment]
fs.inotify.max_user_watches is 8192 on my Ubuntu 14.04 server, so I'm surprised you're hitting limits. I guess some other process is consuming most of the user watches.

**Comment by OnkelTem** [comment]
Maybe it's the docker process, watching everything inside a container? Just a wild guess...
